api_tests/nodes/views/test_node_storage.py

```diff
@@ -0,0 +1,95 @@
+import pytest
+
+from api.base.settings.defaults import API_BASE
+from api.caching import settings as cache_settings
+from api.caching.utils import storage_usage_cache
+from osf_tests.factories import (
+    ProjectFactory,
+    AuthUserFactory
+)
+from osf.utils.permissions import READ, WRITE
+from website import settings
+from website.project.utils import sizeof_fmt
+
+
+@pytest.mark.django_db
+@pytest.mark.enable_quickfiles_creation
+class TestNodeStorage:
+    @pytest.fixture()
+    def admin_contributor(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def write_contributor(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def read_contributor(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def non_contributor(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def project(self, admin_contributor, write_contributor, read_contributor):
+        project = ProjectFactory(
+            creator=admin_contributor
+        )
+        project.add_contributor(write_contributor, WRITE, visible=False)
+        project.add_contributor(read_contributor, READ)
+        project.save()
+        return project
+
+    @pytest.fixture()
+    def url(self, project):
+        return '/{}nodes/{}/storage/'.format(API_BASE, project._id)
+
+    @pytest.fixture()
+    def embed_url(self, project):
+        return '/{}nodes/{}/?embed=storage'.format(API_BASE, project._id)
+
+    def test_node_storage(self, app, url, embed_url, project, admin_contributor,
```

Comment:
nit: similar to above, splitting this into more tests means we get more information about failure modes earlier.